src/utils/rangeMatchModifiers.ts

```diff
@@ -0,0 +1,123 @@
+import { defaultDateLib, type DateLib } from "../classes/DateLib.js";
+import type { DateRange, Matcher } from "../types/index.js";
+
+import { dateMatchModifiers } from "./dateMatchModifiers.js";
+import { rangeIncludesDate } from "./rangeIncludesDate.js";
+import {
+  isDateAfterType,
+  isDateBeforeType,
+  isDateInterval,
+  isDateRange,
+  isDatesArray,
+  isDayOfWeekType
+} from "./typeguards.js";
+
+/**
+ * Returns whether a date range matches against at least one of the given
+ * {@link Matcher}.
+ *
+ * ```tsx
+ * const range: DateRange = {
+ *  from: new Date(2021, 12, 21),
+ *  to: new Date(2021, 12, 30)
+ * }
+ * const matcher1: Date = new Date(2021, 12, 21)
+ * const matcher2: DateRange = {
+ *  from: new Date(2022, 5, 1),
+ *  to: new Date(2022, 5, 23)
+ * }
+ * const dateMatchModifiers(date, [matcher1, matcher2]); // true, since matcher1 is in the date.
+ * ```
+ *
+ * @group Utilities
+ */
+export function rangeMatchModifiers(
+  range: DateRange,
```

Comment:
I think this should help to remove the `range.from`, `range.to` checks inside this function:

```suggestion
  range: Required<DateRange>,
```

If `<Required>` utils does not work as expected, let be even more explicit:

```suggestion
  range: { from: Date, to: Date },
```

Reply:
Thanks for the info 👍 
I thought about doing the same but preferred to get feedback about what pattern to follow for the codebase.
Do you think we should extract this type to the general `types` file in the lib?
Name it something like `FullDateRange` or `CompleteDateRange`.
This could make the typing exported by the lib confusing, so this type could be used only internally.

Reply:
I'd prefer keep it explicit in the argument... but I can apply these style changes once merged - do not worry! Thanks for asking.